modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/ConsensusState.scala

```diff
@@ -1,111 +1,81 @@
 package org.tessellation.node.shared.infrastructure.consensus
 
 import cats.Show
-import cats.kernel.Eq
-import cats.syntax.eq._
-import cats.syntax.option._
 import cats.syntax.show._
 
 import scala.concurrent.duration.FiniteDuration
 
-import org.tessellation.node.shared.infrastructure.consensus.declaration.kind.{PeerDeclarationKind, _}
-import org.tessellation.node.shared.infrastructure.consensus.trigger.ConsensusTrigger
+import org.tessellation.node.shared.infrastructure.consensus.declaration.PeerDeclaration
 import org.tessellation.schema.peer.PeerId
 import org.tessellation.security.hash.Hash
-import org.tessellation.security.signature.Signed
 
 import derevo.cats.{eqv, show}
 import derevo.circe.magnolia.{decoder, encoder}
 import derevo.derive
+import monocle.Lens
+import monocle.macros.GenLens
+
+@derive(eqv, encoder, decoder)
+case class Facilitators(value: List[PeerId])
```

Comment:
Refined classes that are meant to represent certain concepts in the codebase. Hence if a function needs `Facilitators` and `WithdrawnFacilitators` which are supposed to mean the same thing across the codebase then only these field can be extracted (with e.g. a lens) from the state rather than the whole state classes would need to be passed. In the future I would also customize the `ConsensusState` class and `ConsensusResources` class and separate currency and global (and other future consensuses) even more. This is a step in this direction.